pkg/fileutils/reflink_linux.go

```diff
@@ -0,0 +1,20 @@
+package fileutils
+
+import (
+	"io"
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+// LinkOrCopy attempts to reflink the source to the destination fd.
+// If reflinking fails or is unsupported, it falls back to io.Copy().
+func LinkOrCopy(src, dst *os.File) error {
```

Comment:
`ReflinkOrCopy`? I would read this as “hard link or copy”, which is frequently undesirable.

Reply:
agree, `ReflinkOrCopy()` sounds better